src/firefly/js/charts/ui/CombineChart.jsx

```diff
@@ -256,12 +256,16 @@ function combineChart(chartIds, props={}) {
             });
         }
 
-        // failed when only one of the two are mapped(??).  apply map to both(workaround)
         const xmapped = (data?.[activeTrace]?.x || '').startsWith?.('tables:');
         const ymapped = (data?.[activeTrace]?.y || '').startsWith?.('tables:');
         if (xmapped ^ ymapped) {
-            if (!ymapped) set(data, `${activeTrace}.y`, `tables::${tablesources?.[activeTrace]?.mappings?.y}`);
-            if (!xmapped) set(data, `${activeTrace}.x`, `tables::${tablesources?.[activeTrace]?.mappings?.x}`);
+            // all mapped fields should be handled: x, y, error_y.array, etc.
+            Object.entries(tablesources?.[activeTrace]?.mappings || {}).forEach(([key, val]) => {
+                // need get to handle compound keys, like 'error_y.array'
+                if (!get(data, `${activeTrace}.${key}`)?.startsWith?.('tables:')) {
```

Comment:
We're trying to do way with lodash's `get`.  Can you use optional chaining (?.) like you did already?
`!data?.[activeTrace]?.[key]?.startsWith.('tables:')`

Reply:
In table source `mappings` the key is compound: "error_y.array", while in `data` it is not compound, it is a path to the field.